zulip_bots/zulip_bots/test_lib.py

```diff
@@ -108,11 +108,14 @@ def call_request(self, message, expected_method, response, state_handler):
         instance = self.MockClass.return_value
 
         # Set mock identity of the bot
-        name = "test bot"
-        pad = '**' if ' ' in name else ''
-        instance.bot_identity.return_value = (
-            ExternalBotHandler.BotIdentity(name, "{}{}{}".format(pad, name, pad), None))
+        if hasattr(self.__class__, 'bot_identity'):
+            bot_identity = self.bot_identity
```

Comment:
I find this line a bit confsuing; I think we can drop it and instead do
```
if hasattr(self.__class__, 'bot_identity'):
    self.bot_identity = ExternalBotHandler.BotIdentity(name, "{}{}{}".format(pad, name, pad), None)
instance.bot_identity.return_value = self.bot_identity
```
?

Reply:
I didn't follow this immediately, but upon further inspection I understand what you mean :+1: